Add process for election of Rust Foundation Project Directors (#12)
rust-lang/leadership-council: roles/rust-foundation-project-director.md

# Role Description: Rust Foundation Project Director

Please refer to [Rust Foundation Director Roles & Responsibilities](TODO) for components of the role description that apply to all Foundation Board Directors.

> ## Goals

Reviewer: I think we're risking making this proposal much hard to approve if we tie defining what exactly the role of the Project Directors is to how we select them. I realize that it's hard to advertise the role to others if we can't really define what the role is, but alas that's sort of the situation we find ourselves in.

I would prefer that we keep this list short and fill it with things that are extremely unlikely to change over time and leave the last line as a call for helping shape the role together with the Council and the Foundation Staff. For example:

- Represent the needs and interests of the Rust Project at large within the Rust Foundation board
- Provide additional perspective to the Rust Foundation staff on how Foundation activities may impact the Rust Project
- Work with the Rust Leadership Council and Rust Foundation staff on refining the role of Project Director for the future

Author: Am I right in assuming there's an implicit pushback to the "acting as a conduit of information" item implied by this comment? I'm getting that because your list verbatim included my first bullet but didn't include the second. I'm okay with that just want to be sure that's what you meant.

Reviewer: Yes, I think that that particular phrasing is a bit more aspirational than we can likely commit to, and I’d in general like to focus on more concrete examples of what the role entails. So for example instead of “conduit of information”, state that the project directors meet with the staff.

Reviewer 2: > I think we're risking making this proposal much hard to approve if we tie defining what exactly the role of the Project Directors is to how we select them. I realize that it's hard to advertise the role to others if we can't really define what the role is, but alas that's sort of the situation we find ourselves in.

I wonder if it'd be better to separate the role description into a separate PR? Maybe we should have a time limit, like if we don't have consent on the role description by next Tuesday then we pull it out and consent to the election process by itself. When we start the Setup phase, we can either point folks at the role description PR or the finished version if we've agreed on it by then.

Author: I do like that idea, for now I'm going to go ahead and create separate checkboxes for the two documents so we can move ahead with the procedure part right away once we're happy with the state of the document.

Reviewer 3: Chiming in:

> Ensure communication between the Rust Project and Foundation is robust and high fidelity.

I think this is perfect. We should be giving our directors high level goals and let them come up with a strategy they and the staff feel good about. (In less official channels, we should also share our biggest hits and misses so they learn from our mistakes!)

- Represent the needs and interests of the Rust Project at large within the Rust Foundation board
- Act as a conduit of information between the Rust Project and the Rust Foundation

## Time Budget

> - 2-5 hours per week

Reviewer 4: Is this realistic? I'm not sure I would say that over a year I see time averaging out to 5 hours/week, even though some weeks are that high (and on rare occasions higher).

Author: Yes, I believe so. It's based on the surveying Paul and Abi had to do back in sept of last year for our 990 filing. The email was titled "Average time spent by Board members working on the Rust Foundation". The chair reported spending 5 hours per week, other members averaged around 2.5 hours per week. This happened to be right around the time I was heavily using pomodoros for time tracking so I actually had accurate timecard data of how much time I spent on foundation work (minus zulip time) which came out to 19 hours over 3 months or approximately 1.5 hours per week.

Reviewer 4: Hm, okay. I wonder how much bias there is in terms of not doing "too little work" when asked how much work you're doing :)

But I think that helps explain the 5 hours and the rest falls down naturally from there, so that seems okay to me.

> ## Tasks / Activities / Responsibilities

Reviewer: Same point as above. Let's keep this list short. fill it with things we know won't change (e.g., attending Rust Foundation board meetings, meeting with the Rust Foundation staff on a regular basis), and then end it with a TODO of helping shape what this role is in the future.

Reviewer 2: Here's my attempt at a diff that adds @rylev's suggestion. I changed meeting with Foundation staff to as needed rather than regular. My reasoning was that I don't know that we should expect Project Directors to need a standing meeting with various staff members, but they should have open communication lines that they use when issues come up that need to be discussed.

I also added communicating with the Rust Project as an explicit item. We might not know exactly what this looks like, but they are supposed to represent the Project on the Foundation Board, so we need two way communication between the two.

```suggestion
## Tasks / Activities / Responsibilities

* Attend Rust Foundation board meetings
* Meeting with Rust Foundation staff as needed
* Communicate with the Rust Project. Project directors have leeway in how to do this, but at a minimum it should include discussing with the Leadership Council issues that are relevant to the Project.
```

Author: I wish I hadn't forgot to push the diff I had already done, duplicated work, feelsbadman ;_;

## Qualities of a Council member (required skills)

- Strong understanding of the Rust Project, its teams, processes, and ecosystem.

> - Ability to prioritize others' needs, representing all viewpoints from the Rust Project.

Reviewer: I really like these [lines from the Council RFC](https://github.com/rust-lang/rfc-leadership-council/blob/63a867ee7a14599e864b4ccba3964a9f086ae400/text/3392-leadership-council.md?plain=1#L219C7-L224). Can we take those and adjust them to this role?

Author: should be good now, marking as resolved

# Role Description: Rust Foundation Project Director

> Please refer to [Rust Foundation Director Roles & Responsibilities](TODO) for components of the role description that apply to all Foundation Board Directors.

Reviewer 2: Is the linked document here supposed to be the same as the one referenced below, https://foundation.rust-lang.org/static/board-director-role-description.pdf? If not, has this document been written yet?

Reviewer: I believe it is. @yaahc can confirm?

## Goals

- Represent the needs and interests of the Rust Project at large within the Rust Foundation board
- Ensure communication between the Rust Project and Foundation is robust and high fidelity.

> - TODO Fill out this list

Reviewer 2: ```suggestion
```

Is there anything else we need in this list for the first time around?

## Time Budget

- 2-5 hours per week

## Tasks / Activities / Responsibilities

- [Abide by the Roles & Responsibilites of Foundation Board Members](https://foundation.rust-lang.org/static/board-director-role-description.pdf)
- Attend Rust Foundation board meetings
- Help prepare and present the "Project Director Update" for each board meeting
- Meet with the Rust Foundation staff on a regular basis
- Work with the Rust Leadership Council and Rust Foundation staff on refining the role of Project Director for the future

> - TODO Fill out this list

Reviewer 2: ```suggestion
```

## Qualities of a Council member (required skills)

- Sufficient time and energy to dedicate to the needs of the Rust Foundation and the Rust Project.
- An interest in helping with the topics of Project operations and Project governance.
- A keen sense of the needs of Rust Project, its teams, processes, and ecosystem.
- Willingness to understand and represent viewpoints they may not have personal experience with or necessarily agree with.

> - TODO Fill out this list

Reviewer 2: ```suggestion
```

Please refer to [Rust Foundation Director Roles & Responsibilities](TODO) for components of the role description that apply to all Foundation Board Directors.

## Goals

- Represent the needs and interests of the Rust Project at large within the Rust Foundation board
- Ensure communication between the Rust Project and Foundation is robust and high fidelity.
- TODO Fill out this list

## Time Budget

- 2-5 hours per week

## Tasks / Activities / Responsibilities

- [Abide by the Roles & Responsibilites of Foundation Board Members](https://foundation.rust-lang.org/static/board-director-role-description.pdf)
- Attend Rust Foundation board meetings
- Help prepare and present the "Project Director Update" for each board meeting
- Meet with the Rust Foundation staff on a regular basis
- Work with the Rust Leadership Council and Rust Foundation staff on refining the role of Project Director for the future
- TODO Fill out this list

## Qualities of a Council member (required skills)

- Sufficient time and energy to dedicate to the needs of the Rust Foundation and the Rust Project.
- An interest in helping with the topics of Project operations and Project governance.

> - Broad awareness of the needs of the Project outside of their teams or areas of active contribution.

Reviewer 3: This one seems redundant with the next one. And I'm not sure how I would define "broad awareness" if I were trying to assess if I were a good fit. A lot of that awareness comes with the job, and the council will be there to assist. So I'd argue that while a director does need some sense of the needs of the Project, the breadth comes from being willing to consider the needs and viewpoints of others (see below).

Author: ```suggestion
```

- A keen sense of the needs of Rust Project, its teams, processes, and ecosystem.

> - Ability to prioritize others' needs, representing all viewpoints from the Rust Project, not just those they agree with.

Reviewer 3: Hmm, I think a director should be prepared to express their opinion on a matter as well (at least one that the council has not taken an official position on, but maybe even then). I think the role is more than just a liaison.

I also think the skill applies to more than just members of the project; it applies to other directors as well as foundation staff members and prominent community members.

This feels closer to the mark for me, though it could be improved..

```suggestion
- Willingness to empathize with, and express, views that they may not have personal experience with or necessarily agree with.
```

Author: yea I'm not feeling thrilled with the wording but I can't figure out how to describe my objection, it just feels like pure vibes, like preferring the word "viewpoints" to "views" and "understand" over "empathize with", but I think those are ultimately coming from my viewpoint, lol, and the things these words mean to me. How "viewpoints" is centering people rather than ideas, and how "understanding" is centering the goal of representation, whereas "empathizing" to me sounds to me like it could boil down to "so and so is upset about this" without really understanding _why_ they're upset.

In retrospect that actually wasn't that hard to figure out... lol

```suggestion
- Willingness to understand and represent viewpoints they may not have personal experience with or necessarily agree with.
```

Reviewer 2: I noticed in the reword that we lost "from the Rust project." Do we want to keep that?

Reviewer 3: I'm happy with the wording @yaahc proposed.

> I noticed in the reword that we lost "from the Rust project." Do we want to keep that?

That was deliberate on my part:

> I also think the skill applies to more than just members of the project; it applies to other directors as well as foundation staff members and prominent community members.
